src/System Application/App/Company Initialize/CompanyInitializeImpl.Codeunit.al

```diff
@@ -0,0 +1,51 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Environment.Configuration;
+
+using System.Environment;
+
+codeunit 901 "Company Initialize Impl."
+{
+    Access = Internal;
+
+    procedure InitializeCompany()
+    var
+        CompanyInitialize: Record "Company Initialize";
+        CompanyInitializeCodeunit: Codeunit "Company Initialize";
+        ModuleInfo: ModuleInfo;
+    begin
+        CompanyInitialize."Initialized Time" := CurrentDateTime();
+
+        if NavApp.GetCurrentModuleInfo(ModuleInfo) then
+            CompanyInitialize."Initialized Version" := format(ModuleInfo.AppVersion());
+        CompanyInitialize.Insert();
```

Comment:
We can initialize company many times, the codeunit 2 must be safe to run many times.   This logic is also wrong, it should be moved after Initialize company not before. We could partially commit and keep the record.